### PR TITLE
Fix vmap mid roll ad position reporting

### PR DESCRIPTION
**Pull request by strangesource**
## Problem
Ad position reporting for client side ad insertion with VMAP and Google IMA ads integration was reporting wrong ad position for mid- and post-roll ads.

## Changes
As the timing information on the ad is not reported correctly for VMAP ads (tracked internally), we take the correct time from the current active ad break. This is in line with the tacking on web.

### CODE REVIEW

File: conviva/src/test/kotlin/com/bitmovin/analytics/conviva/ConvivaAnalyticsIntegrationTest.kt [hunk 1]
```diff
@@ -186,6 +191,31 @@ class ConvivaAnalyticsIntegrationTest {
...
+        }
+
+        player.listeners[PlayerEvent.AdBreakStarted::class]?.forEach {
+            it(createAdBreakStartedEvent(00.0))
+        }
+
+        verify { videoAnalytics.reportAdBreakStarted(any(), any()) }
+        player.listeners[PlayerEvent.AdStarted::class]?.forEach { it(TEST_AD_STARTED_EVENT) }
+        verify {
+            adAnalytics.reportAdStarted(match { it["c3.ad.position"] == AdPosition.PREROLL })
```
Comment (krocard) [inline conviva/src/test/kotlin/com/bitmovin/analytics/conviva/ConvivaAnalyticsIntegrationTest.kt:214]: I didn't know this `match` helper. Nice!

**Review by krocard — APPROVED** [review]

File: ConvivaTestApp/src/androidTest/java/com/bitmovin/analytics/conviva/testapp/AdvertisingTests.kt [hunk 2]
```diff
@@ -90,4 +90,64 @@ class AdvertisingTests {
...
+                    }
+            )
+            player
+        }
+
+        mainHandler.postWaiting { player.load(Sources.Dash.basicLive) }
+
+        // mid-roll ad break is scheduled at 15 seconds, will play immediately in case of the DASH
+        // live stream due to absolute time stamps.
+        player.expectEvent<PlayerEvent.AdBreakStarted> {
```
[the remark this answers is not in the document]
Comment (strangesource) [reply ConvivaTestApp/src/androidTest/java/com/bitmovin/analytics/conviva/testapp/AdvertisingTests.kt:148]: [7ee0bd3](https://github.com/bitmovin/bitmovin-player-android-analytics-conviva/pull/83/commits/7ee0bd34b39a86df99f7e5ca503fe47b2619a843) I went for a mock to make it easier to read.

File: conviva/src/test/kotlin/com/bitmovin/analytics/conviva/ConvivaAnalyticsIntegrationTest.kt [hunk 3]
```diff
@@ -235,3 +265,49 @@ private val attachedPlayerEvents = listOf(
...
+            override val height: Int
+                get() = 100
+            override val id: String?
+                get() = null
+            override val isLinear: Boolean
+                get() = true
+            override val mediaFileUrl: String?
+                get() = null
+            override val width: Int
+                get() = 200
```
[the remark this answers is not in the document]
Comment (strangesource) [reply conviva/src/test/kotlin/com/bitmovin/analytics/conviva/ConvivaAnalyticsIntegrationTest.kt]: [7ee0bd3](https://github.com/bitmovin/bitmovin-player-android-analytics-conviva/pull/83/commits/7ee0bd34b39a86df99f7e5ca503fe47b2619a843) I went for a mock to make it easier to read.

File: conviva/src/test/kotlin/com/bitmovin/analytics/conviva/ConvivaAnalyticsIntegrationTest.kt [hunk 4]
```diff
@@ -186,6 +191,31 @@ class ConvivaAnalyticsIntegrationTest {
         verify(exactly = 0) { videoAnalytics.reportPlaybackError(any(), any()) }
     }
 
+    @Test
+    fun `reports CSAI ad position based on last ad break schedule time`() {
+
```
Comment (matamegger) [inline conviva/src/test/kotlin/com/bitmovin/analytics/conviva/ConvivaAnalyticsIntegrationTest.kt:196]:
```suggestion
```

File: conviva/src/test/kotlin/com/bitmovin/analytics/conviva/ConvivaAnalyticsIntegrationTest.kt [hunk 5]
```diff
@@ -186,6 +191,31 @@ class ConvivaAnalyticsIntegrationTest {
...
+        }
+
+        verify { videoAnalytics.reportAdBreakStarted(any(), any()) }
+        player.listeners[PlayerEvent.AdStarted::class]?.forEach { it(TEST_AD_STARTED_EVENT) }
+        verify {
+            adAnalytics.reportAdStarted(match { it["c3.ad.position"] == AdPosition.MIDROLL })
+        }
+
+        player.listeners[PlayerEvent.AdBreakStarted::class]?.forEach {
+            it(createAdBreakStartedEvent(00.0))
```
Comment (matamegger) [inline conviva/src/test/kotlin/com/bitmovin/analytics/conviva/ConvivaAnalyticsIntegrationTest.kt:208]:
```suggestion
            it(createAdBreakStartedEvent(0.0))
```

**Review by matamegger — APPROVED** [review]
Looks good!
Besides some style nits, nothing to add to the other review.

**Pull request merged**